common/signals.py

```diff
@@ -1,9 +1,21 @@
+import logging
+import time
+
 from django.contrib.auth.models import User
+from django.contrib.auth.signals import user_logged_in, user_login_failed
 from django.db.models.signals import post_save, pre_save
 from django.dispatch import receiver
 
 from common.models import EmailVerificationTable, UserProfile
 
+logger = logging.getLogger(__name__)
+
+
+@receiver(user_login_failed)
+def log_failed_attempt(sender, credentials: dict, request=None, **kwargs):
+    logger.debug("User login attempt failed. Sleeping for 1 sec.")
+    time.sleep(1)
```

Comment:
Won't that freeze the whole webserver for one second? Or is that intentional (I would think that way, but worth asking:))

Reply:
It will block the current thread but on second thought I will remove this since we have the brute force protection anyhow.